base/server/src/main/java/com/netscape/cms/password/PasswordChecker.java

```diff
@@ -44,7 +162,11 @@ public PasswordChecker() {
      *         returns false.
      */
     public boolean isGoodPassword(String pwd) {
-        return pwd != null && pwd.length() >= MIN_LEN;
+        try {
+            return checkPassword(pwd) == PasswordQuality.CMS_PASSWORD_GOOD;
+        } catch (EPasswordCheckException e) {
```

Comment:
We only have a few places that call isGoodPassword now.  Could we maybe just throw an exception with the reason instead, and get rid of getReason()? See comment for getReason() below.